src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala

```diff
@@ -341,19 +331,23 @@ private class AppTaskLauncherActor(
 
   private[this] def handleTaskOp(taskOp: TaskOp, offer: Mesos.Offer): Unit = {
     def updateActorState(): Unit = {
+      val taskId = taskOp.taskId
       taskOp match {
         // only decrement for launched tasks, not for reservations:
         case _: TaskOp.Launch => tasksToLaunch -= 1
         case _                => ()
       }
 
-      val taskId = taskOp.taskId
-      taskOp.maybeNewTask match {
-        case Some(newTask) =>
-          tasksMap += taskId -> newTask
+      // We will receive the updated task once it's been persisted. Before that,
+      // we can only store the possible state, as we don't have the updated task
+      // yet.
+      taskOp.newTask.possibleNewState match {
+        case Some(newState) =>
+          tasksMap += taskId -> newState
           scheduleTaskOpTimeout(taskOp)
+
         case None =>
-          removeTask(taskId)
```

Comment:
hmmm. "removeTask" is not really needed -- it was mostly defensive and maybe misleading. It was different in some historical state, I think. But then, we can actually just not do anything because the task should not be in the tasksMap. This branch is unnecessary, isn't it?